operator/controllers/common/phasehandler.go

```diff
@@ -106,3 +113,68 @@ func (r PhaseHandler) HandlePhase(ctx context.Context, ctxTrace context.Context,
 
 	return &PhaseResult{Continue: false, Result: requeueResult}, nil
 }
+
+func (r PhaseHandler) createFailureReasonSpanEvents(ctx context.Context, phase common.KeptnPhaseType, object *PhaseItemWrapper) (string, error) {
+	if phase.IsEvaluation() {
+		return r.GetEvaluationFailureReasons(ctx, phase, object)
+	} else if phase.IsTask() {
+		return r.GetTaskFailureReasons(ctx, phase, object)
+	}
+	return "", nil
```

Comment:
Here, I would add a warning log that smth is off and we call the method in an unexpected way

Reply:
resolved